ARKit-QRScanner/QRScanner.swift

```diff
@@ -11,59 +11,303 @@ import ARKit
+                }
+            }
+            if points.count < 4 {return}
+            
+            var planeT = computeBestFitPlane(points: points)
+            let quat = simd_quatf(from: simd_float3(x: 0, y: 1, z: 0), to: simd_float3(x: 0, y: 0, z: 1))
+            let normalT = matrix_float4x4(quat)
+            let planeInXZ = simd_mul(planeT, normalT)
+            
+            
+            // The plane is the xz-plane of the local coordinate space this transform defines.
+            let bottomLeft = frame.camera.unprojectPoint(bottomLeftInView, ontoPlane: planeInXZ, orientation: orientation, viewportSize: viewSize) ?? bottomLeftRough
+            let bottomRight = frame.camera.unprojectPoint(bottomRightInView, ontoPlane: planeInXZ, orientation: orientation, viewportSize: viewSize) ?? bottomRightRough
+            let topRight = frame.camera.unprojectPoint(topRightInView, ontoPlane: planeInXZ, orientation: orientation, viewportSize: viewSize) ?? topRightRough
+            let topLeft = frame.camera.unprojectPoint(topLeftInView, ontoPlane: planeInXZ, orientation: orientation, viewportSize: viewSize) ?? topLeftRough
+            
+            
+            
+            let positionRough = simd_float3(planeT.columns.3.x, planeT.columns.3.y, planeT.columns.3.z)
+            let position = frame.camera.unprojectPoint(centerInView, ontoPlane: planeInXZ, orientation: orientation, viewportSize: viewSize) ?? positionRough
+            planeT.columns.3 = simd_float4(position, 1)
+            
+            let size = CGSize(width: CGFloat(simd_distance(bottomLeft, bottomRight) * 0.5 + simd_distance(topLeft, topRight) * 0.5), height: CGFloat(simd_distance(bottomLeft, topLeft) * 0.5 + simd_distance(topRight, bottomRight) * 0.5))
+            
+            
+            let response = QRResponse(feature: feature, position: position, transform: planeT, accuracy: .transformAndSizeApprox, size: size)
+            
+            responses.append(response)
+        }
+    }
+    
+    /// Return a list of QR Codes and transform, size
+    ///
+    /// - Parameter view: ARSCNView provided by ARKit
+    /// - Parameter viewSize: ARSCNView's size
+    /// - Returns: A QRResponse array containing the estimated transform, feature and accuracy
+    @available(iOS 12.0, *)
+    public static func findQRTransform(view:ARSCNView ,viewSize:CGSize) -> [QRResponse] {
+        guard let capturedImage = view.session.currentFrame?.capturedImage else { return [QRResponse]() }
+        let features = QRScanner.findQR(in: capturedImage)
+        let imageWidth = CGFloat(CVPixelBufferGetWidth(capturedImage))//1920
+        let imageHeight = CGFloat(CVPixelBufferGetHeight(capturedImage))//1440
+        
+        let factorH = viewSize.height / imageWidth
+        let factorW = viewSize.width / imageHeight
+        let factor = CGFloat.minimum(factorH, factorW)
+        
+        
+        
+        return features.reduce(into: [QRResponse]()) { (responses, feature) in
```

Comment:
Looks like this is repeated code of the previous function. Could you take out this functionality and put it into a private method?

Reply:
OK